### ISSUE TITLE
the server could not find the requested resource (post datasetsinternal.com.ie.ibm.hpsys)

### ISSUE THREAD

**Issue by vmasyagin** [datashim-io/datashim#201]
Good afternoon. Installed DLF on kubernetes version 22 using kubectl apply -f https://raw.githubusercontent.com/datashim-io/datashim/master/release-tools/manifests/dlf.yaml. Everything started, but when I create a dataset
```
cat <<EOF | kubectl apply -f-
apiVersion: com.ie.ibm.hpsys/v1alpha1
kind: dataset
metadata:
  name: s3-dataset
spec:
  local:
    type: "COS"
    accessKeyID: "{AWS_ACCESS_KEY_ID}"
    secretAccessKey: "{AWS_SECRET_ACCESS_KEY}"
    endpoint: "{S3_SERVICE_URL}"
    bucket: "{BUCKET_NAME}"
    readonly: "true" #OPTIONAL, default is false
    region: "" #OPTIONAL
EOF
```
 then I get an error:
```
ERROR controller.dataset Reconciler error {"reconciler group": "com.ie.ibm.hpsys", "reconciler kind": "Dataset", "name": "s3-dataset", "namespace": "default", " error": "the server could not find the requested resource (post datasetsinternal.com.ie.ibm.hpsys)"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
 /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
 /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
```
What could be the reason?

**Comment by vmasyagin** [comment]
It's because of the start order. Start order is important.

**Comment by srikumar003** [comment]
Thanks @vmasyagin. Did this happen because the dataset-operator pods were still starting at the time you created a dataset ?

**Comment by vmasyagin** [comment]
Here is an interesting point. I put it on a cluster without the Internet. When I applied the dlf.yaml file, then, of course, the images could not be downloaded. I uploaded the images to the local repository and began to change images sequentially in daemon sets, deployments and stateful sets. After that I created a dataset and saw the error described above. I demolished everything, immediately changed the images to local ones in dlf.yaml, applied it and everything works fine.
